src/main/java/seedu/weme/storage/JsonSerializableRecords.java

```diff
@@ -0,0 +1,59 @@
+package seedu.weme.storage;
+
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonRootName;
+
+import seedu.weme.commons.exceptions.IllegalValueException;
+import seedu.weme.model.Records;
+import seedu.weme.model.RecordsManager;
+
+/**
+ * An Immutable Records that is serializable to JSON format.
+ */
+@JsonRootName(value = "records")
+public class JsonSerializableRecords {
+
+
+    private Set<String> pathRecords;
+    private Set<String> descriptionRecords;
+    private Set<String> tagRecords;
+    private Set<String> nameRecords;
+
+    /**
+     * Constructs a {@code JsonSerializableRecords} with the given records details.
+     */
+    @JsonCreator
+    public JsonSerializableRecords(@JsonProperty("pathRecords") Set<String> pathRecords,
+                                    @JsonProperty("descriptionRecords") Set<String> descriptionRecords,
+                                    @JsonProperty("tagRecords") Set<String> tagRecords,
+                                    @JsonProperty("nameRecords") Set<String> nameRecords) {
+
+        this.pathRecords = pathRecords;
+        this.descriptionRecords = descriptionRecords;
+        this.tagRecords = tagRecords;
+        this.nameRecords = nameRecords;
+    }
+
+    /**
+     * Converts a given {@code Records} into this class for Jackson use.
+     */
+    public JsonSerializableRecords(Records records) {
+        pathRecords = records.getPaths();
+        descriptionRecords = records.getDescriptions();
+        tagRecords = records.getTags();
+        nameRecords = records.getNames();
+    }
+
+    /**
+     * Converts this serializable records object into the model's {@code Records} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the serializable records.
+     */
+    public Records toModelType() throws IllegalValueException {
+        return new RecordsManager(pathRecords, descriptionRecords, tagRecords, nameRecords);
```

Comment:
likewise for here, use a new set of records. Don't forget to add records to weme's `toModelType()`